participants/l10n_update_ts.py

```diff
@@ -226,22 +236,28 @@ def init_gitdir(self, reponame):
                     'has_downloads': False,
                     'auto_init': True
                 }
-                ghresp = requests.post("%s/user/repos" % self.gitconf["apiurl"],
-                                       auth=gitserv_auth,
-                                       headers={
-                                           'content-type': 'application/json'
-                                       },
-                                       verify=False,
-                                       data=json.dumps(payload))
+                ghresp = requests.post(
+                    "%s/user/repos" % self.gitconf["apiurl"],
+                    auth=gitserv_auth,
+                    headers={'content-type': 'application/json'},
+                    verify=False,
+                    data=json.dumps(payload)
+                )
                 ghresp.raise_for_status()
 
-            check_call(["git", "clone",
-                        self.gitconf["repourl"] % {
-                            "username": self.gitconf["username"],
-                            "reponame": reponame
-                        }],
+            repo_url = self.gitconf["repourl"] % {
+                "username": self.gitconf["username"],
+                "reponame": reponame
+            }
+            check_call(["git", "clone", repo_url, gitdir],
                        cwd=self.gitconf["basedir"])
+            current_branch = check_output(
+                ['git', 'symbolic-ref', '-q', 'HEAD'], cwd=gitdir).strip()
+            if current_branch.replace('refs/heads/', '') != branch:
+                check_call(["git", "branch", branch], cwd=gitdir)
+                check_call(["git", "checkout", branch], cwd=gitdir)
+            check_call(["git", "reset", "--hard", "origin/%s" % branch],
+                       cwd=gitdir)
         else:
             check_call(["git", "fetch"], cwd=gitdir)
-            check_call(["git", "rebase", "origin/master"], cwd=gitdir)
-        return gitdir
+            check_call(["git", "rebase", "origin/%s" % branch], cwd=gitdir)
```

Comment:
Given a fail here is retried as a clean/clone/checkout then I wonder is this really supposed to be a rebase? I'd expect a checkout -f or reset --hard to the branch along with similar branch existence checks.

Reply:
Yep, that's true. And now that I think about it, if there are some local extra commits (possibly leftover from earlier failures?), they should be thrown away, not rebased.